sound/soc/soc-dai.c

```diff
@@ -391,6 +391,45 @@ bool snd_soc_dai_stream_valid(struct snd_soc_dai *dai, int dir)
 	return stream->channels_min;
 }
 
+/*
+ * snd_soc_dai_link_set_capabilities() - set dai_link properties based on its DAIs
+ */
+void snd_soc_dai_link_set_capabilities(struct snd_soc_dai_link *dai_link)
+{
+	struct snd_soc_dai_link_component *cpu;
+	struct snd_soc_dai_link_component *codec;
+	struct snd_soc_dai *dai;
+	bool supported_direction;
+	int direction;
+	int i;
+
+	for_each_pcm_streams(direction) {
```

Comment:
you could get rid of this external loop by verifying both directions on each iteration of internal loops, but I'm not sure that can be done without making it look uglier.

Reply:
added a small array to avoid direction-based tests, looks more compact to me.